puppet/modules/jenkins_job_builder/files/theforeman.org/pipelines/test/foreman-installer.groovy

```diff
@@ -0,0 +1,77 @@
+pipeline {
+    agent none
+    options {
+        timeout(time: 1, unit: 'HOURS')
+        ansiColor('xterm')
+    }
+
+    stages {
+        stage('Test') {
+            matrix {
+                agent any
+                axes {
+                    axis {
+                        name 'ruby'
+                        values '2.5'
+                    }
+                }
+                environment {
+                    PUPPET_VERSION = '6.0'
+                }
+                stages {
+                    stage('Setup Git Repos') {
+                        steps {
+                            ghprb_git_checkout()
+                            sh "cp Gemfile Gemfile.${ruby}-${PUPPET_VERSION}"
+                        }
+                    }
+                    stage("Setup RVM") {
+                        steps {
+                            configureRVM(ruby, "${ruby}-${PUPPET_VERSION}")
+                        }
+                    }
+                    stage('Install dependencies') {
+                        steps {
+                            withRVM(["bundle install --gemfile=Gemfile.${ruby}-${PUPPET_VERSION}"], ruby, "${ruby}-${PUPPET_VERSION}")
+                        }
+                    }
+                    stage('Run Rubocop') {
+                        steps {
+                            withRVM(["BUNDLE_GEMFILE=Gemfile.${ruby}-${PUPPET_VERSION} bundle exec rake rubocop TESTOPTS='-v' --trace"], ruby, "${ruby}-${PUPPET_VERSION}")
+                        }
+                    }
+                    stage('Run Tests') {
+                        steps {
+                            withRVM(["BUNDLE_GEMFILE=Gemfile.${ruby}-${PUPPET_VERSION} bundle exec rake spec TESTOPTS='-v' --trace"], ruby, "${ruby}-${PUPPET_VERSION}")
+                        }
+                    }
+                    stage('Test installer configuration') {
+                        steps {
+                            script {
+                              install_dir = pwd(tmp: true)
+                            }
+                            sh "rm -rf _build/"
+                            withRVM(["BUNDLE_GEMFILE=Gemfile.${ruby}-${PUPPET_VERSION} bundle exec rake build PREFIX=${install_dir} --trace"], ruby, "${ruby}-${PUPPET_VERSION}")
```

Comment:
Given the BUNDLE GEMFILE is needed in many lines, worth wrapping with [withEnv](https://www.jenkins.io/doc/pipeline/steps/workflow-basic-steps/#withenv-set-environment-variables)?

Reply:
I'll look into that now that I have it fully working.